tincr/io/library/genCellLibrary.tcl

```diff
@@ -879,10 +987,25 @@ proc ::tincr::create_xml_cell_library { {part xc7a100t-csg324-3} {filename ""} {
         puts $xml_out "    </cell>"
         remove_cell $cell_instance -quiet
     }
-
+    
     create_port_xml $site_map $xml_out
     
     puts $xml_out "  </cells>"
+    
+    # Create the xml for each macro cell supported in the device
+    puts "Creating macro definitions..."
+    set supported_macros [tincr::get_supported_macro_cells]
+    
+    if {[llength $supported_macros] > 0} {
+        puts $xml_out "  <macros>"
+        foreach macro_lib_cell $supported_macros {
+            set macro_cell [create_cell -reference $macro_lib_cell "tmp" -quiet]
+            tincr::write_macro_xml $macro_cell $xml_out
+            remove_cell $macro_cell -quiet
```

Comment:
If the command is only supposed to run under certain conditions (e.g. no currently open design), then we should restrict the user from calling it when those conditions are not met. I see a couple options here:
1. Add a guard statement like you suggested
1. Don't export this command, effectively making it a "private method" (scoped to this namespace) and call it from another proc that ensures that the conditions are satisfied